File: tests/java/tests-java-gradle-junit5-allure3-retrofit2/src/test/java/tests/api/BackendWiringApiTests.java
package tests.api;

import annotations.Layer;
import api.ApiTestBase;
import api.Calls;
import io.qameta.allure.Epic;
import io.qameta.allure.Feature;
import io.qameta.allure.Severity;
import io.qameta.allure.SeverityLevel;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;

/**
 * Deployed-stand facts about the active backend module and its PostgreSQL wiring.
 * Response shapes/types are the api layer's job ({@code HealthItemsApiTests}).
 */
@Layer("api")
@Epic("Wired backend")
@Feature("Health and data source")
@Severity(SeverityLevel.BLOCKER)
@DisplayName("Backend wiring on deployed stand")
class BackendWiringApiTests extends ApiTestBase {

    @Test
    @Tag("api")
    @Tag("smoke")
    @DisplayName("GET /api/health — deployed service is the active backend module, not a neighbour")
    void healthReportsActiveBackendService() {
        var response = Calls.execute(api.health());
        assertEquals(200, response.code());
        assertNotNull(response.body());
        assertEquals(config.apiHealthService(), response.body().service());
    }

    @Test
    @Tag("api")
    @DisplayName("GET /api/items — catalogue is served from PostgreSQL, not a stub or fallback")
    void itemsAreWiredToPostgreSQL() {
        var response = Calls.execute(api.items());
        assertEquals(200, response.code());
        assertNotNull(response.body());
        assertEquals("postgresql", response.body().source());
    }
}

File: tests/csharp/tests-csharp-xunit-allure3-playwright/Tests/Api/BackendWiringApiTests.cs
using Allure.Net.Commons;
using Allure.Net.Commons.Attributes;
using Api;
using RestSharp;
using Tests;

namespace Tests.Api;

/// <summary>
/// Deployed-stand facts about the active backend module and its PostgreSQL wiring.
/// Response shapes/types are the api layer's job (<c>HealthItemsApiTests</c>).
/// </summary>
[AllureEpic("Wired backend")]
[AllureFeature("Health and data source")]
[AllureSeverity(SeverityLevel.blocker)]
[AllureSuite("Backend wiring on deployed stand")]
public sealed class BackendWiringApiTests : ApiTestBase
{
    [Trait("TestCategory", "api")]
    [Trait("TestCategory", "smoke")]
    [Fact(DisplayName = "GET /api/health — deployed service is the active backend module, not a neighbour")]
    public void HealthReportsActiveBackendService()
    {
        var response = RestSharpHttp.Request(Method.Get, "/api/health");
        Assert.True(Equals(200, response.Status), response.Body);
        Assert.Equal(Config.ApiHealthService, response.Text("service"));
    }

    [Trait("TestCategory", "api")]
    [Fact(DisplayName = "GET /api/items — catalogue is served from PostgreSQL, not a stub or fallback")]
    public void ItemsAreWiredToPostgreSQL()
    {
        var response = RestSharpHttp.Request(Method.Get, "/api/items");
        Assert.True(Equals(200, response.Status), response.Body);
        Assert.Equal("postgresql", response.Text("source"));
    }
}
